forkable/forkable_test.go

```diff
@@ -1026,6 +1035,16 @@ func TestForkable_ProcessBlock(t *testing.T) {
 			result, err := json.MarshalIndent(p.results, "", "  ")
 			require.NoError(t, err)
 
+			for i := range p.results {
+				if c.expectedResult[i].reorgJunctionBlock == nil {
+					assert.Nil(t, p.results[i].reorgJunctionBlock)
+				} else if p.results[i].reorgJunctionBlock == nil {
+					assert.Nil(t, c.expectedResult[i].reorgJunctionBlock)
+				} else {
+					assert.Equal(t, c.expectedResult[i].reorgJunctionBlock.String(), p.results[i].reorgJunctionBlock.String())
+				}
```

Comment:
With not just `assert.Equal(t, c.expectedResult[i].reorgJunctionBlock, p.results[i].reorgJunctionBlock)`?

Reply:
Because they are two different types of BlockRef, so they are not equal. Their representation under String() is equal, though, but I can't call String() on a nil interface without panic, and panics are harder to read in test results...

Reply:
```suggestion
expected := c.expectedResult[i].reorgJunctionBlock
actual := p.results[i].reorgJunctionBlock

assert.True(t, bstream.EqualsBlockRefs(expected, actual), "%s != %s", expected, actual)
```